### PR TITLE
[CRN-153] team member inactive date

### PR DESCRIPTION
**Pull request by ricardocarvalh0**
https://asaphub.atlassian.net/browse/CRN-1281

### CODE REVIEW

File: apps/crn-server/test/entities/event.test.ts [hunk 1]
```diff
@@ -205,6 +205,7 @@ describe('events entity', () => {
                 },
               ],
               role: 'Lead PI (Core Leadership)',
+              inactiveSinceDate: null,
```
Comment (peterstarling) [inline apps/crn-server/test/entities/event.test.ts:208]: can you add a test where this date is actually returned? or maybe add this date to the fixtures rather than having a null

File: packages/squidex/schema/crn/schemas/users.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     },
     "scripts": {
       "create": "if(JSON.stringify(ctx.data.teams.iv).indexOf('ASAP Staff') > 0  && ctx.data.role.iv !== 'Staff') {\n    reject('Only Staff can be given this team role. Change ASAP Hub Role to \"Staff\".')\n}",
-      "update": "if(JSON.stringify(ctx.data.teams.iv).indexOf('ASAP Staff') > 0  && ctx.data.role.iv !== 'Staff') {\n    reject('Only Staff can be given this team role. Change ASAP Hub Role to \"Staff\".')\n}"
+      "update": "if(JSON.stringify(ctx.data.teams.iv).indexOf('ASAP Staff') > 0  && ctx.data.role.iv !== 'Staff') {\n    reject('Only Staff can be given this team role. Change ASAP Hub Role to \"Staff\".')\n}\n\nconst becameAlumni = ctx.data.alumniSinceDate.iv && (ctx.oldData.alumniSinceDate.iv !== ctx.data.alumniSinceDate.iv);\nlet anyTeamChanged = false;\n\nctx.data.teams.iv = ctx.data.teams.iv.map((team) => {\n    if (becameAlumni && !team.inactiveSinceDate) {\n        team.inactiveSinceDate = new Date().toISOString();\n        anyTeamChanged = true;\n    }\n    return team;\n});\n\nif (anyTeamChanged) {\n    replace();\n}"
```
Comment (peterstarling) [inline packages/squidex/schema/crn/schemas/users.json:12]: there is no way to test it really so i assume this has been manually tested and it works? was it copied from somewhere?

File: apps/crn-server/src/entities/team.ts [hunk 3]
```diff
@@ -39,6 +39,10 @@ export const parseGraphQLTeamMember = (
     ?.filter((t) => t.id && t.id[0]?.id === teamId)
     .filter((s) => s.role)[0]?.role;
 
+  const inactiveSinceDate = user.flatData.teams
```
Comment (peterstarling) [inline apps/crn-server/src/entities/team.ts:42]: shouldnt there be a test for this somewhere?

**Review by peterstarling — APPROVED** [review]

File: packages/squidex/schema/crn/schemas/users.json (hunk 2 above)
Comment (ricardocarvalh0) [reply packages/squidex/schema/crn/schemas/users.json:12]: correct, it was tested and copied from https://cloud.squidex.io/app/crn-2528/schemas/users?tab=scripts

File: apps/crn-server/test/fixtures/users.fixtures.ts [hunk 4]
```diff
@@ -154,6 +155,19 @@ export const getGraphQLUser = (
           },
         ],
       },
+      {
```
Comment (peterstarling) [inline apps/crn-server/test/fixtures/users.fixtures.ts:158]: to be honest with you i don't think you should add another team just to test one property (which is essentially what you are doing). Why don't you simply write a single test where the team has the `inactiveSinceDate` property instead?

File: apps/crn-server/test/handlers/webhooks/fetch-by-code/fetch-by-code.test.ts [hunk 5]
```diff
@@ -151,7 +151,12 @@ describe('Fetch-user-by-code handler', () => {
       )) as APIGatewayProxyResult;
 
       expect(result.statusCode).toStrictEqual(200);
-      expect(JSON.parse(result.body)).toMatchObject(getUserResponse());
+
+      // Using stringify() in the expected response since the result body
+      // is also using it and undefined properties are removed when running it.
+      const expected = JSON.parse(JSON.stringify(getUserResponse()));
```
Comment (peterstarling) [inline apps/crn-server/test/handlers/webhooks/fetch-by-code/fetch-by-code.test.ts:157]: you say that this needs to be done because of undefined properties but `toMatchObject` should allow you to have them

**Review by peterstarling — CHANGES_REQUESTED** [review]

File: apps/crn-server/test/fixtures/users.fixtures.ts (hunk 4 above)
Comment (ricardocarvalh0) [reply apps/crn-server/test/fixtures/users.fixtures.ts:158]: sorry my intention was to reproduce a user that is part of two teams and is active in one and inactive in the other one

Comment (peterstarling) [reply apps/crn-server/test/fixtures/users.fixtures.ts:158]: cool, so then remove this second team :D

File: apps/crn-server/test/handlers/webhooks/fetch-by-code/fetch-by-code.test.ts [hunk 6]
```diff
@@ -151,7 +151,11 @@ describe('Fetch-user-by-code handler', () => {
       )) as APIGatewayProxyResult;
 
       expect(result.statusCode).toStrictEqual(200);
-      expect(JSON.parse(result.body)).toMatchObject(getUserResponse());
+
+      // Using stringify() in the expected response since the result body
```
Comment (peterstarling) [inline apps/crn-server/test/handlers/webhooks/fetch-by-code/fetch-by-code.test.ts:155]: remove comment

**Review by peterstarling — APPROVED** [review]
